### PR TITLE
asm.var.summary should be on if the screen is too small ##visual

### PR DESCRIPTION
**Pull request by Vane11ope**
about this #13259

but still not thoroughly done probably and so much room to improve so you'd better keep this issue open just yet.

Screenshot from 2019-03-29 02-32-38

Screenshot from 2019-03-29 02-33-15

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -1685,6 +1685,17 @@ static void ds_show_functions(RDisasmState *ds) {
 	ds->stackptr = core->anal->stackptr;
 	RAnalFcnVarsCache vars_cache;
 	r_anal_fcn_vars_cache_init (core->anal, &vars_cache, f);
+
+	int o_varsum = ds->show_varsum;
```
Comment (radare) [inline libr/core/disasm.c]: i think this is bool

File: libr/core/disasm.c [hunk 2]
```diff
@@ -1685,6 +1685,17 @@ static void ds_show_functions(RDisasmState *ds) {
...
+	if (!ds->show_varsum) {
+		int numvars = vars_cache.bvars->length + vars_cache.rvars->length + vars_cache.svars->length;
+		if (numvars > ds->l) {
```
Comment (radare) [inline libr/core/disasm.c]: if ds->l s the amount of lines of the screen.. maybe we want to do osome -5 or -10 for padding

Comment (radare) [reply libr/core/disasm.c]: also i think we should only do this when in interactive mode, for visual

Comment (Vane11ope) [reply libr/core/disasm.c]: i agree with both, 30 mins

Comment (Vane11ope) [reply libr/core/disasm.c]: better 10 i guess for Vpp

Comment (Vane11ope) [reply libr/core/disasm.c]: done.

**Comment by radare** [comment]
waiting for travis, but it should be green

**Comment by Vane11ope** [comment]
cool, take your time...with travis :)

**Pull request merged**